src/Runner.Worker/ActionManifestManager.cs

```diff
@@ -476,14 +480,41 @@ public ActionDefinitionData Load(IExecutionContext executionContext, string mani
                     }
                     else
                     {
+                        var compositeActionsExecutionDataSteps = steps.Cast<Pipelines.ActionStep>().ToList();
+                        if (defaultsToken == null)
+                        {
+                            foreach (var step in compositeActionsExecutionDataSteps)
+                            {
+                                if (step.Reference.Type != Pipelines.ActionSourceType.Script)
+                                {
+                                    continue;
+                                }
+                                var mapping = step.Inputs.AssertMapping("inputs");
+                                var foundShell = false;
+                                foreach (var pair in mapping)
+                                {
+                                    // Expression key
+                                    if (pair.Key is StringToken && pair.Key.AssertString("inputs key").Value.Equals("shell"))
```

Comment:
[minor] instead of using the `foundShell` var can we use negation on this line and throw directly the error without the extra assignments to `foundShell`?

Reply:
How can I do that? I'm going through all mappings trying to validate that the `shell` is present. Only if no mapping exists for shell, I need to throw

Reply:
Was thinking of something like
```
foreach (var pair in mapping)
{
	if !(pair.Key is StringToken && pair.Key.AssertString("inputs key").Value.Equals("shell"))
	{
		throw new TemplateValidationException("Composite action must set the default shell, or the shell parameter in each step is required.");
	}
}
```
but I can live without this 😃